main.tf

```
@@ -675,172 +259,172 @@ resource "aws_iam_user" "s3_bucket_iam_user" {
 }
 
 resource "aws_iam_policy" "s3_bucket_with_kms_iam_policy_1" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? 1 : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 ? 1 : 0
 
   name        = "${var.iam_user_policy_name}-S3BucketObjectPolicy"
   policy      = data.aws_iam_policy_document.s3_bucket_with_kms_policy_document_1[0].json
   description = "Policy for bucket and object permissions when a KMS alias is specified"
 }
 
 resource "aws_iam_user_policy_attachment" "attach_s3_bucket_with_kms_iam_policy_1" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? var.number_of_users : 0
+  count = local.use_kms_encryption && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 ? var.number_of_users : 0
 
   user       = aws_iam_user.s3_bucket_iam_user[count.index].name
   policy_arn = aws_iam_policy.s3_bucket_with_kms_iam_policy_1[0].arn
 }
 
 resource "aws_iam_policy" "s3_bucket_with_kms_iam_policy_2" {
-  count = var.kms_alias != "" && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 && var.website_hosting == "false" ? 1 : 0
+  count = var.kms_alias != "" && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 && ! var.website_hosting ? 1 : 0
```

```suggestion
  count = local.use_kms_encryption && length(var.whitelist_ip) == 0 && length(var.whitelist_vpc) == 0 ? 1 : 0
```

You changed a couple of the policies and attachments above to use the new variable, but there are still quite a few places where it could be used below (I'm not highlighting the others since I didn't want to be too repetitive). This is pretty much just for consistancy though.

---

eagle-eyed Tasharn...
I've amended to use `local.use_kms_encryption` where it can be used